assets/js/modules/adsense/components/module/ModuleOverviewWidget/Overview.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * ModuleOverviewWidget component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { calculateChange } from '../../../../../util';
+import { Grid, Row, Cell } from '../../../../../material-components';
+import DataBlock from '../../../../../components/DataBlock';
+
+const Overview = ( {
+	metrics,
+	currentRangeData,
+	previousRangeData,
+	selectedStats,
+	handleStatsSelection,
+} ) => {
+	const { totals, headers } = currentRangeData;
+	const { totals: previousTotals } = previousRangeData;
+
+	const dataBlocks = [
+		{
+			className: 'googlesitekit-data-block--page-rpm googlesitekit-data-block--button-1',
+			title: metrics[ headers[ 0 ].name ],
+			datapoint: totals[ 0 ],
+			datapointUnit: headers[ 0 ]?.currency,
+			change: calculateChange( previousTotals[ 0 ], totals[ 0 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 0,
+			handleStatsSelection,
+		},
+		{
+			className: 'googlesitekit-data-block--page-rpm googlesitekit-data-block--button-2',
+			title: metrics[ headers[ 1 ].name ],
+			datapoint: totals[ 1 ],
+			datapointUnit: headers[ 1 ]?.currency,
+			change: calculateChange( previousTotals[ 1 ], totals[ 1 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 1,
+			handleStatsSelection,
+		},
+		{
+			className: 'googlesitekit-data-block--impression googlesitekit-data-block--button-3',
+			title: metrics[ headers[ 2 ].name ],
+			datapoint: totals[ 2 ],
+			change: calculateChange( previousTotals[ 2 ], totals[ 2 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 2,
+			handleStatsSelection,
+		},
+		{
+			className: 'googlesitekit-data-block--impression googlesitekit-data-block--button-4',
+			title: metrics[ headers[ 3 ].name ],
+			datapoint: totals[ 3 ],
+			datapointUnit: '%',
+			change: calculateChange( previousTotals[ 3 ], totals[ 3 ] ),
+			changeDataUnit: '%',
+			context: 'button',
+			selected: selectedStats === 3,
+			handleStatsSelection,
+		},
+	];
+
+	return (
+		<Grid>
+			<Row>
+				{ dataBlocks.map( ( block, i ) => (
+					<Cell
+						key={ i }
+						smSize={ 2 }
+						mdSize={ 2 }
+						lgSize={ 3 }
+					>
+						<DataBlock
```

Comment:
We only have 4 stats and this is fixed so let's keep them them inline (no iteration necessary).

The wrapping `Cell` props can be extracted to a variable instead to avoid duplicating those for each one since they are always the same (`key` was the one different and that's no longer needed if we remove the map) 🙂

Reply:
@aaemnnosttv I had to remove the `Cell` component to instead use custom class names. I think there was a valid reason to use the custom class names before especially in cases we have very long numbers, which the custom classes take care of. So the new widget is similar to the legacy one. About using `map`, I prefer if we keep it like that, because it reduces the markup and everything is cleaner IMHO. Also, I think it'll be consistent about how we display the blocks on other modules `Overview` components as well.